### PR TITLE
[sdks] Add a XCODE32_DIR make variable and use it to compile the 32 bit cross compilers.

### PR DESCRIPTION
**Pull request by lewurm**
Forward port of https://github.com/mono/mono/pull/10139

Needed for `arm64_32` (https://github.com/mono/mono/pull/11691): We still need to build AOT compilers that are 32bit, which is only possible with Xcode9. However, `arm64_32` needs Xcode10.

### CODE REVIEW

File: sdks/versions.mk [hunk 1]
```diff
@@ -23,7 +23,9 @@ ANDROID_SDK_VERSION_x86_64?=21
 
 # iOS
 
-XCODE_DIR?=/Applications/Xcode.app/Contents/Developer
+XCODE_DIR?=/Applications/Xcode101.app/Contents/Developer
+# Xcode version used to compile 32 bit cross compilers
+XCODE32_DIR?=/Applications/Xcode94.app/Contents/Developer
```
Comment (lewurm) [inline sdks/versions.mk]: This maps to the configuration we have on our CI machines: https://github.com/mono/mono/pull/11691#issuecomment-439178459

File: sdks/versions.mk [hunk 2]
```diff
@@ -23,7 +23,9 @@ ANDROID_SDK_VERSION_x86_64?=21
 
 # iOS
 
-XCODE_DIR?=/Applications/Xcode.app/Contents/Developer
+XCODE_DIR?=/Applications/Xcode101.app/Contents/Developer
```
Comment (luhenry) [inline sdks/versions.mk]: I would keep the directory name to Xcode.app and follow the convention that it is always the latest stable Xcode (10.1 in the current situation)

Comment (lewurm) [reply sdks/versions.mk]: Okay, I've moved the settings needed for the bots to `run-jenkins.sh`. Is that better?

Comment (akoeplinger) [reply sdks/versions.mk]: what if we make `Xcode.app` point to the latest one and `Xcode32bit.app` to the one which can build x86 apps? then we don't need to hardcode version numbers in the scripts anywhere and just rely on bots being set up correctly.

Comment (lewurm) [reply sdks/versions.mk]: we can do that later. for now, if we would do that, it would break current master.

Comment (akoeplinger) [reply sdks/versions.mk]: Not if you add another symlink for the old name, but fair enough :)

**Comment by lewurm** [comment]
@monojenkins build failed

**Comment by lewurm** [comment]
@monojenkins squash

**Comment by akoeplinger** [comment]
Merging since this only affects the SDKs and those passed (the rest hit some Jenkins issue).

**Comment by lewurm** [comment]
@monojenkins backport 2018-10
@monojenkins backport 2018-12

**Pull request merged**